docs/rules/detect-circular-reference.md

```diff
@@ -0,0 +1,55 @@
+---
+description: 'Detects usage of `forwardRef()` function commonly used to handle circular references'
+---
+
+[Forward references](https://docs.nestjs.com/fundamentals/circular-dependency#forward-reference) are commonly used to handle circular dependencies between services or modules. For example, if `FooService` and `BarService` depend on each other, you can use `forwardRef()` to resolve the circular dependency. However, `forwardRef()` must be a last resort, and we generally recommend changing your code to avoid circular dependencies. This rule detects usage of the `forwardRef()` method so you can keep track of what potentially needs refactoring.
+
+One strategy to avoid circular dependencies between services is to make each responsible for a single use case. That way you decrease the interface and likelihood of circular dependencies. You can also avoid using services as facades for data repositories. Instead, services should be usually used to encapsulate business logic (commands). Leave query responsibilities to repositories.
```

Comment:
I'd be hesitant to advise detailed alternative strategies. It could easily be interpreted as _the_ official way to structure a Nest app, whereas it's highly dependent on the context. i.e. if it's a simple REST app, entity-based module structure is appropriate, and simply splitting splitting the service would probably suffice.

If the app's primary responsibility is transforming and loading data, transaction scripts are a great fit. 

Highly complex apps with lots of business logic would again require a different approach, and DDD would probably be a better fit, but that would depend on the skills within the team too 😄 

I'd suggest we drop the second paragraph and just stick to the generic "we recommend refactoring the code to avoid the circular dependency" from the first paragraph 🙂

Reply:
Yeah, I think you are right; I thought of providing some direction to users who would ask themselves, "Then what should we do?", but maybe a blog post is a better way of doing it